### ISSUE TITLE
[pt] Disambiguator improvement: SPS00 to DA - 2022-10-01

### ISSUE THREAD

**Issue by marcoagpinto** [languagetool-org/languagetool#7150]
Hello @jaumeortola 

Take a look at these two sentences:
```
Tom tem dinheiro suficiente para comprar a casa que quer.
Tom tem dinheiro suficiente para comprar o carro que quer.
```

In the first one, “comprar a casa” turns “a” into SPS00.

In the second one, “comprar o carro” turns “o” into DA blah blah.

Could the disambiguator be improved to show also “a” as DA blah blah?

Thanks!

**Comment by jaumeortola** [comment]
I resolved the issue for this sentence. But the disambiguation of `a` (SPS00 vs. DA0FS0) is quite hard. Let's leave the issue open, and collect more examples of wrong disambiguation for `a`.

**Comment by marcoagpinto** [comment]
@jaumeortola 

I am almost certain that if you have an infinitive verb followed by an “a” it returns SPS00.

Or maybe infinitive + “a” + noun


So, it is easier to fix than it seems, at least for this case.

**Comment by marcoagpinto** [comment]
I had a busy night fixing false positives in my thesis, like you can see for the numerous commits.

❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️ ❤️

**Comment by jaumeortola** [comment]
> I am almost certain that if you have an infinitive verb followed by an “a” it returns SPS00.

Yes. There is a subrule of A_PREPOSTION doing exactly that (with some exceptions), and it has a comment that says it is "too aggressive".

So, with `infinitive + “a” + noun` or even `verb + “a” + noun`, is `a` always a determiner?

**Comment by marcoagpinto** [comment]
> So, with `infinitive + “a” + noun` or even `verb + “a” + noun`, is `a` always a determiner?

```
Vamos comprar a casa.
Vamos comprar o carro.
```

It seems so.

**Comment by marcoagpinto** [comment]
I am not sure if it is only with infinite.

**Comment by marcoagpinto** [comment]
but with infinite it seems to happens.

**Comment by marcoagpinto** [comment]
When I say nouns, maybe I should also say adjectives?
